third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint

```diff
@@ -31,8 +31,15 @@
 #include "src/FixedPoint/FixedPointTypes.h"
 
 // Use optimized implementations whenever available
-#ifdef EIGEN_VECTORIZE_AVX2
+#ifdef EIGEN_VECTORIZE_AVX512
+#include "src/Tensor/TensorContractionThreadPool.h"
+#include "src/FixedPoint/PacketMathAVX512.h"
+#include "src/FixedPoint/TypeCastingAVX512.h"
+
+#elif defined EIGEN_VECTORIZE_AVX2
```

Comment:
Why elif? It seems so sad to go from AVX2 to AVX-512 only to loose the optimized AVX2 matmul. Can't we fiddle the packet_traits definitions to enable us to use the optimized quantized AVX2 matmul even when AVX512 is enabled for other things like casting and elementwise ops? (Fine to do as a followup I suppose)